ci/build.py

```diff
@@ -71,7 +71,7 @@ def get_docker_binary(use_nvidia_docker: bool) -> str:
 
 
 def build_docker(platform: str, docker_binary: str, registry: str, num_retries: int, no_cache: bool,
-                 cache_intermediate: bool) -> str:
+                 cache_intermediate: bool = False) -> str:
```

Comment:
This has to be true.

Reply:
I don't mind changing it to `True`, but the existing behaviour before introducing the flag was `False`.

Reply:
One issue I noted during local development when setting this to `True`, is that my disk can fill up very quickly because Docker will not clean any of the cached layers automatically.

Reply:
Well there has been some back and forth between refactors - maybe somebody removed it :/. Initially, I designed it that way that intermediary layers were kept since that's the whole point of the partial cache. This test validates that assumption: https://github.com/apache/incubator-mxnet/blob/master/ci/test_docker_cache.py#L184

Note that this test suite is not run in CI...

Reply:
Well you can't have everything - either partial caching and manual cleanup or automatic cleanup but no partial caching. I think it is fine to expect that people use docker purge every now and then.

Reply:
Does the CI docker purge automatically? My docker cache quickly grow by 200GB after developing locally with the intermediate caching enabled. It's fine to purge time to time manually, but it's better to automate this.

Reply:
> Note that this test suite is not run in CI...

Is it possible to run this suite in CI? If so, would you like to help enable it?

Reply:
For reference, you removed the `rm=false` yourself in f107397b753ff08bb19e7572bc1a9ebedd832f88

Reply:
CI does not docker purge automatically. But since the lifecycle of unix slaves is very very short, it's not an issue. Also, in 99% of the cases, no new layers are built and only the cache is used, so the likelyhood of a single slave regenerating a lot of new mismatching layers is quite low. On a dev machine though, that's a different case.

Reply:
Yeah it should be possible. It will just cause issues if the test fails since that might mean that stuff gets left dangling, but due to the mentioned lifecycle, that shouldn't have a big impact.

Feel free to enable it.

Reply:
Good catch! I can't fully recall what was the thing around --rm=false, but I recall that I wrote tests to cover all these unknowns. Rule of thumb is that if the tests pass, we're good to go (given they use the same parameters of course).

Reply:
> CI does not docker purge automatically. But since the lifecycle of unix slaves is very very short, it's not an issue

If the intermediate containers are only kept on the slave, and we only generate them in PRs that change the Dockerfile, I think we don't need to use them at all on CI. `--rm=false` then seems only useful for development.

Reply:
Yep exactly. But please make sure that the tests pass before merging - otherwise we might break the partial caching.

Reply:
This PR does not change the status quo. There is no risk.